pan-domain-auth-verification/src/main/scala/com/gu/pandomainauth/service/CryptoConf.scala

```diff
@@ -0,0 +1,44 @@
+package com.gu.pandomainauth.service
+
+import com.gu.pandomainauth.SettingsFailure.SettingsResult
+import com.gu.pandomainauth.service.Crypto.keyFactory
+import com.gu.pandomainauth.service.CryptoConf.SettingsReader.{privateKeyFor, publicKeyFor}
+import com.gu.pandomainauth.{InvalidBase64, MissingSetting, PublicKeyFormatFailure}
+import org.apache.commons.codec.binary.Base64.{decodeBase64, isBase64}
+
+import java.security.spec.{InvalidKeySpecException, PKCS8EncodedKeySpec, X509EncodedKeySpec}
+import java.security.{PrivateKey, PublicKey}
+import scala.util.Try
+
+
+
+object CryptoConf {
+  case class SettingsReader(settingMap: Map[String,String]) {
+    def setting(key: String): SettingsResult[String] = settingMap.get(key).toRight(MissingSetting(key))
+
+    val activePublicKey: SettingsResult[PublicKey] = setting("publicKey").flatMap(publicKeyFor)
+
+    def activeKeyPair: SettingsResult[KeyPair] = for {
+      publicKey <- activePublicKey
+      privateKey <- setting("privateKey").flatMap(privateKeyFor)
+    } yield KeyPair(publicKey, privateKey)
+  }
+
+  object SettingsReader {
+    def publicKeyFor(data: Array[Byte]) = keyFactory.generatePublic(new X509EncodedKeySpec(data))
```

Comment:
there are two public definitions of `publicKeyFor` (and `privateKeyFor`) in this object -- just wanted to check that both are meant to be exposed publicly? What's the expected use-case for a consumer to use `def publicKeyFor(data: Array[Byte])` as opposed to the `publicKeyFor(base64EncodedKey: String)` method?

Reply:
Yeah, good point - the overloading of method names was a bit confusing - addressed with 4c87946a113b73f23e9e9b683082bdaeb49a5e2d!